docs/usage.rst

```diff
@@ -189,6 +189,62 @@ Since version ``0.20.0``, the ``pika-bind`` helper script is bundled with pika.
 and then runs the given command. ``pika-bind`` is a more convenient alternative to manually setting ``PIKA_PROCESS_MASK``
 when pika is used together with a runtime that may reset the process mask of the main thread, like OpenMP.
 
+.. _logging:
+
+Logging
+=======
+
+The pika runtime logs warnings and more severe messages to stderr by default, using `spdlog
+<https://github.com/gabime/spdlog>`_. To change the logging level, set the ``PIKA_LOG_LEVEL``
+environment variable to a value between 0 (trace) and 6 (off) (the values correspond to levels in
+spdlog). The log messages are sent to stderr by default. The destination can be changed by setting
+the ``PIKA_LOG_DESTINATION`` environment variable. Supported values are:
+
+- ``cerr``
+- ``cout``
+- any other value is interpreted as a path to a file
+
+pika will by default print messages in the following format:
+
+.. code-block::
+
+   [2024-04-18 13:45:07.095279283] [pika] [info] [host:machine(-1)] [pid:2786603] [tid:2786607] [pool:0000/0003/0003] [parent:----/----] [task:0x7fa6a4077cf0/pika_main] [set_thread_state.cpp:205:set_thread_state] set_thread_state: thread(0x7fa6a802c8d0), description(<unknown>), new state(pending), old state(suspended)
+
+The fields are as follows:
+
+- ``[2024-04-18 13:45:07.095279283]``: The timestamp of the message.
+- ``[pika]``: An identifier present in all pika's logs.
+- ``[info]``: The severity level of the message.
+- ``[host:machine(-1)]``: The hostname and the MPI rank of the process (``-1`` if MPI is disabled).
```

Comment:
Is it too complex/slow to just have `[hostname]` instead of `[hostname(-1)]` when MPI is not used?

Reply:
But I see below that `%j` is set to the MPI rank, for custom logging. So an empty string might be surprising to some users.

Reply:
I updated the "hostname" field to be `[host:<hostname>/----]` in the case that the MPI rank is unavailable, with the motivation that the output is more predictable in the sense that _something_ is always printed in the MPI rank field, but not the magic number `-1`. This is consistent with how I've treated the other fields so far (task description/id, thread pool ids, etc.), i.e. indicate that the field exists in general, but isn't available at that moment.

@RMeli do you think this is ok, or do you think it's more confusing to have these "empty fields"?